### PR TITLE
Add username errors to /login response

### PR DESCRIPTION
**Pull request by teddylo57**
A list of errors will be returned on /login if there are any issues in the format ```{"errors":{"errorList":["Username cannot contain DEALER.","Username cannot start with a digit."]}}```

### CODE REVIEW

File: backend/src/main/java/poker/models/websocket/ErrorResponse.java [hunk 1]
```diff
@@ -2,14 +2,15 @@
...
+
 @Data
 public class ErrorResponse {
 
   private Errors errors;
 
   @Data
   public class Errors {
-
-    private String username;
```
Comment (teddylo57) [inline backend/src/main/java/poker/models/websocket/ErrorResponse.java:13]: Removed username field as I think it is no longer necessary, if you want we can add the username sent for a more complete message but I think it is redundant

File: backend/src/main/java/poker/Routes.java [hunk 2]
```diff
@@ -29,21 +31,22 @@ private void userRoutes() {
...
           }
         });
   }
 
-  private String loginError(String error) {
+  private String loginError(List<String> errors) {
     ErrorResponse errorResponse = new ErrorResponse();
     ErrorResponse.Errors usernameError = errorResponse.new Errors();
-    usernameError.setUsername(error);
+    usernameError.setErrorList(errors);
```
Comment (ericliu121187) [inline backend/src/main/java/poker/Routes.java:49]: Could the response model be:

```json
{
  errors: {
     username: String[], 
  }
}
```

File: backend/src/main/java/poker/services/users/UserService.java [hunk 3]
```diff
@@ -256,10 +257,21 @@ public static boolean isValid(UUID uuid, String username) {
         && !Character.isDigit(username.charAt(0))
         && playersMap.values().stream()
             .filter(player -> !player.getUid().equals(uuid))
-            .noneMatch(player -> player.getUsername().toLowerCase().equals(username.toLowerCase()))
-        && playersMap.values().stream()
+            .noneMatch(player -> player.getUsername().toLowerCase().equals(username.toLowerCase()));
+  }
+
+  public static List<String> getLoginErrors(UUID uuid, String username) {
```
Comment (ericliu121187) [inline backend/src/main/java/poker/services/users/UserService.java:263]: `validateLogin` might be a more descriptive, I've seen that as convention.

Comment (teddylo57) [reply backend/src/main/java/poker/services/users/UserService.java:263]: Fixed

**Comment by teddylo57** [comment]
Example message: 

```{
  "errors": {
    "username": [
      "Username cannot contain DEALER.",
      "Username cannot start with a digit."
    ]
  }
}```

File: backend/src/main/java/poker/models/websocket/ErrorResponse.java [hunk 4]
```diff
@@ -2,14 +2,16 @@
...
 
-  private Errors errors;
-
-  @Data
-  public class Errors {
+  private Map<String, List<String>> errors;
 
-    private String username;
-  }
+//  @Data
```
Comment (ericliu121187) [inline backend/src/main/java/poker/models/websocket/ErrorResponse.java:13]: Is this meant to be removed?

**Pull request merged**